sekoia_automation/scripts/sync_library.py

```diff
@@ -0,0 +1,311 @@
+            return True
+        auth = HTTPBasicAuth(registry_user, registry_pat)
+
+        if match := re.match(r"(.*?)\/(.*)\/(.*)", docker_image):
+            registry_path = match[1]
+            registry_pathinfo = match[2]
+            image_name = match[3]
+        else:
+            registry_path = "ghcr.io"
+            registry_pathinfo = "v2/sekoialab"
+            image_name = docker_image
+
+        response = requests.get(
+            f"https://{registry_path}/token",
+            params={
+                "service": registry_path,
+                "scope": "repository:<repo>:pull",
+                "client_id": "symphony-docker-image",
+            },
+            auth=auth,
+        )
+        token = response.json()["token"]
+
+        response = requests.get(
+            f"https://{registry_path}/{registry_pathinfo}/{image_name}/manifests/{docker_image_version}",
+            headers={
+                "Authorization": f"Bearer {token}",
+                "Accept": "application/vnd.docker.distribution.manifest.v2+json",
+            },
+        )
+
+        return response.status_code == 200
+
+    def load_module(self, module_path: Path) -> None:
+        manifest_path = module_path / "manifest.json"
+
+        with manifest_path.open() as fd:
+            module_info = json.load(fd)
+
+        if self.registry_check and not self.check_image_on_registry(
+            module_info["docker"], module_info["version"]
+        ):
+            raise DockerImageNotFoundError(
+                f"Image {module_info['docker']}:{module_info['version']}\
+    not available on registry"
+            )
+
+        triggers = self.set_docker(self.load_triggers(module_path), module_info)
+        actions = self.set_docker(self.load_actions(module_path), module_info)
+
+        module_uuid: str = module_info["uuid"]
+        module_name: str = module_info["name"]
+        module_image: str | None = self.get_module_logo(module_path)
+
+        if module_image:
+            module_info["image"] = module_image
+
+        self.sync_module(module_info)
+        if triggers:
+            self.sync_list(
+                module_name=module_name,
+                module_uuid=module_uuid,
+                list_objects=triggers,
+                name="trigger",
+            )
+            print()
+        if actions:
+            self.sync_list(
+                module_name=module_name,
+                module_uuid=module_uuid,
+                list_objects=actions,
+                name="action",
+            )
+            print()
+
+    def load(self, library_path: Path) -> None:
+        modules = [
+            p
+            for p in library_path.iterdir()
+            if p.is_dir() and p.joinpath("manifest.json").exists()
+        ]
+        print(f"Loading {len(modules)} modules from library in database\n")
+
+        for module in modules:
+            self.load_module(library_path / module.name)
+
+        print()
```

Comment:
Empty print, not sure why

Reply:
Just to add empty lines for aesthetic purposes I guess? They were already there in the source script

Reply:
Fair enough